0000-interledger-protocol-4/0000-interledger-protocol-4.md

```diff
@@ -19,44 +19,58 @@ Interledger is a protocol for sending packets of money across different payment
-12. Connector validates the sender's signed payment channel update (or other proof of settlement) and increases the sender's balance by the amount they paid. If the sender does not pay for the ILP Prepare, the connector may refuse to process further ILP packets until they do or the connector may block that user entirely.
+7. Receiver checks the ILP Prepare packet, according to whatever is stipulated by the higher-level protocol (such as checking whether the amount received is above some minimum specified by the sender). Receiver also checks that the Prepare would not put the connector's unsettled balance above the maximum the receiver tolerates (this is the same as the minimum balance limit of the connector _with the receiver_).
+8. To accept the packet, the receiver returns an ILP Fulfill packet, containing the preimage of the condition in the ILP Prepare packet. The receiver may include additional data, the formatting of which is determined by the higher-level protocol. If the receiver does not want the ILP Prepare packet or it does not pass one of the checks, the receiver returns an ILP Reject packet.
+9. If the receiver returned an ILP Fulfill, the connector checks that the fulfillment hashes to the original condition and that the receiver returned it before the expiry in the ILP Prepare packet. If so, the connector returns the same ILP Fulfill packet to the previous connector and credits the receiver's account with the amount in the ILP Prepare packet. If the receiver returned an ILP Reject packet or the Prepare expired before the Fulfill was returned, the connector returns an ILP Reject packet to the previous connector (note that connectors SHOULD return ILP Reject packets when the Prepare expires, even if they have not yet received a response from the next node). Each connector repeats this step (crediting the account of the node that returned the Fulfill) until the Fulfill (or Reject) packet reaches the sender.
+10. Sender checks that the preimage in the ILP Fulfill packet matches the condition in their Prepare packet and may read any data returned by the receiver. The sender may keep a local record of the total value of packets fulfilled to determine how much they owe their connector.
+11. Sender repeats the process, starting from step 1, as many times as is necessary to transfer the desired total amount of value.
+
+After each packet is fulfilled, each noce may sign an update to their payment channel with their peer to cover the value of the packet, or they may rebalance their accounts less frequently using slower on-ledger transfers or physical deliveries of cash or other goods. Each node may limit their peers' maximum outstanding balances, and they may automatically credit accounts when payment channel updates or external transfers are received.
```

Comment:
typo: _noce_

I think this is meant to read "node" but I suggest "participant" instead (categorically) as in my earlier comments.

Reply:
> ...they may automatically credit accounts when payment channel updates or external transfers are received.

Suggest rephrasing to active voice:

> ..they may automatically credit accounts when they receive payment channel updates or external transfers.